content/posts/2021-11-21-tăng-tốc.md

```diff
@@ -0,0 +1,65 @@
+
+**Bonus:** [https://stateofjs.com/](https://stateofjs.com/) is a great resource for you to know about the trending in javascript every year
+
+### 4. In active development
+![Team work](/media/teamwork.jpg)
+
+It's very important that's you are choosing an active project instead of a dead/ unmaintained project. The active project gets better and better over time through community feedback. An unmaintained project does not move forward and does not fix functional bugs and patch security issues. Sometimes, a very popular package could get abandoned and go into the "freeze" state with many open issues and pull requests. It's might used to a great solution in the past, but maybe this is the sign for us we have to move on. An example is [https://github.com/jamiebuilds/react-loadable](https://github.com/jamiebuilds/react-loadable). It's a great solution for a very long time to split code in React. I totally love it. But's stale now with many issues and PRs since 2018 (this post is written at the end of 2021). Now, if I need to split code in React, I rather use https://github.com/gregberge/loadable-components, which is in active development, getting more popular, patch bugs reported by the community and most importantly, solve my problems. My personal advice: choose a package that active in the last 3-6 months, with issues are being resolved and PRs are being merged.
+
+### 5. License
+![License](/media/license.jpg)
+
+It's very important that the package you install grant you permission to use it for commercial in production. Packages in the wild are great. But not all of them are free to use. 
+
+Let's take an example. If you are working with charts, you might hear of [Highcharts](https://github.com/highcharts/highcharts) - Rock-solid and incredibly flexible charting library made for developers. You download it from [npm](https://www.npmjs.com/), make a chart, it looks great, and you are ready to roll out the feature. But please note that Highcharts is not free for commercial. So if you are writing code for your company, haven't bought a license and use this package for production, you are basically breaking the law. We don't want that, right? So please buy a license before using Highcharts in production. Myself, I use [Recharts](https://github.com/recharts/recharts) instead, a free package powered by [D3](https://github.com/d3/d3) with MIT license and in very active development.
```

Comment:
- free for commercial => free for commercial use
- your company, haven't bought => your company, and your company haven't bought